libs/dmlf/include/dmlf/execution/execution_result.hpp

```diff
@@ -103,8 +103,17 @@ struct MapSerializer<fetch::dmlf::ExecutionResult, D>
   template <typename Constructor>
   static void Serialize(Constructor &map_constructor, Type const &exec_res)
   {
-    auto map = map_constructor(3);
-    map.Append(OUTPUT, exec_res.output_);
+    auto             map    = map_constructor(3);
+    variant::Variant output = exec_res.output_;
+
+    if (output.IsUndefined())
```

Comment:
the `IsUndefined` check should't be done at `ExecutionResult` serialization, but rather in the `variant::Variant` serializer/deserializer

Reply:
No, it should be done here - it does not belong in the serialisation.

Reply:
in my understanding Undefined is a reflection of the variant being empty which is part of the variant state not the ExecutionResult